kotlinx-coroutines-core/jvm/src/debug/internal/AgentPremain.kt

```diff
@@ -69,3 +69,28 @@ internal object AgentPremain {
         }
     }
 }
+
+/**
+ * This class is loaded if and only if kotlinx-coroutines-core was used as a javaagent argument.
+ * It serves as a "safe" wrapper around [AgentPremain] that does not require any kotlin-stdlib classes to be loaded for its initialization.
+ *
+ * Initialization of [AgentPremain] may fail, e.g., because it started before the required kotlin-stdlib classes were loaded.
+ * If `kotlinx.coroutines.apply.debug.agent` is set to `false`, the coroutine debug agent will not be applied.
+ */
+@Suppress("unused")
+internal object SafeAgentPremain {
```

Comment:
I don't understand the purpose of this wrapper. What's stopping us from adding the check to `AgentPremain` itself? Won't it be enough to call something like this function as the first thing in `AgentPremain.premain`?
```kotlin
private fun checkIfStdlibIsAvailable() {
  try {
    Result.success(42)
  } catch (_: Throwable) {
    error("kotlinx.coroutines debug agent failed to load " +
        "because the Kotlin standard library is not present in the classpath.")
  }
}
```

Reply:
This makes sense, yes) This is the first point when we find out that stdlib is not loaded

Reply:
If someone later adds a field of some stdlib type to the class, it will cause problems at class loading time. The wrapper will protect against that.

Reply:
No, actually this does not help, because the error happens when the `AgentPremain` class is trying to initialize, at the moment `AgentPremain.<clinit>` is called. And this check in `premain` is already too late.